### ISSUE TITLE
AllowedUntrustedInternalConnections does not respect settings in 5.10

### ISSUE THREAD

**Issue by jprusch** [mattermost/mattermost-server#10679]
#### Summary
AllowedUntrustedInternalConnections does not respect settings in 5.10

#### Steps to reproduce
1. Configure IP, hostname or CIDR settings for AllowedUntrustedInternalConnections
2. Restart MM server
3. Open a channel which retrieves avatar images from one of the IPs, hostnames or CIDRs set to be allowed

#### Expected behaviour
The avatars are displayed

#### Observed behavior (that appears unintentional) 
Avatars are not shown, javascript console logs an error 500:
`error fetching remote image: Get https://connect.schaeferbarthold.de/jira/secure/useravatar?size=xsmall&amp;ownerId=rs&amp;avatarId=10500#0x0: Get https://connect.schaeferbarthold.de/jira/secure/useravatar?size=xsmall&amp;ownerId=rs&amp;avatarId=10500: address forbidden, you may need to set AllowedUntrustedInternalConnections to allow an integration access to your internal network
`

What I did:

I connected our Jira server to Mattermost 5.10.0 & configured AllowedUntrustedInternalConnections with the following settings (Five trys...):

1. IP addresses only (192.168.2.190 192.168.2.88)
2. Hostnames only (connect.schaeferbarthold.de chat.schaeferbarthold.de)
3. Mixed (Both of the above)
4. CIDR (192.168.0.0/16)
5. All by just trying a star (*)

I checked, that the settings were written to config.json & restarted the server after each change. 

The mattermost.log varies depending on the settings:

With IP addresses:
`2019-04-23T14:38:01.759+0200    error   log/log.go:296  error fetching remote image: Get https://connect.schaeferbarthold.de/jira/secure/useravatar?size=xsmall&amp;avatarId=10122#0x0: Get https://connect.schaeferbarthold.de/jira/secure/useravatar?size=xsmall&amp;avatarId=10122: address forbidden, you may need to set AllowedUntrustedInternalConnections to allow an integration access to your internal network       {"source": "stdlog"}
`

With hostnames or CIDR:
`2019-04-23T14:27:16.296+0200    error   log/log.go:296  forbidden content-type: "text/html"     {"source": "stdlog"}
`
The image URL by itself works even with an anonymous access to Jira:
https://connect.schaeferbarthold.de/jira/secure/useravatar?size=xsmall&amp;avatarId=10122

I made sure that the mattermost user can retrieve the URL via wget.

Screenshot:

![image](https://user-images.githubusercontent.com/9337414/56584244-ae6af880-65db-11e9-9bb8-8f160064a978.png)

Log:

[mattermost.log](https://github.com/mattermost/mattermost-server/files/3107744/mattermost.log)

Any advice?!?

**Comment by amyblais** [comment]
@svelle / @esethna Do you know if there is a bug here / if this is a known issue?

**Comment by svelle** [comment]
Haven't seen this yet, but will try to look further into it.

**Comment by amyblais** [comment]
Opened a ticket here: https://mattermost.atlassian.net/browse/MM-15365.

**Comment by DSchalla** [comment]
Hi @jprusch,

taking a look at the logs shared above, the following line stands out to me from the hostname example:
```
2019-04-23T14:27:16.296+0200 error log/log.go:296 forbidden content-type: "text/html" {"source": "stdlog"} 
```

This error is thrown from our image proxy (third-party library) that request for the image returned a `text/html` response instead of an image. You've mentioned above that you've confirmed via wget that the URL is reachable, was this tested from the server running Mattermost or your local machine?

**Comment by jprusch** [comment]
Hi @DSchalla 

this was tested on our Mattermost server using the user running the Mattermost service.

**Comment by jprusch** [comment]
Hi all,

maybe the access.log of our Jira server can help on solving the issue.

The image requests from the Mattermost server (192.168.2.88) return different error messages depending on the URL requested:

`192.168.2.88 707x425008x1 - [02/May/2019:11:47:53 +0200] "GET /jira/secure/useravatar?size=xsmall&amp;avatarId=10122 HTTP/1.1" 404 2293 16 "-" "mattermost-5.10.0" "-"
192.168.2.88 707x425010x1 - [02/May/2019:11:47:53 +0200] "GET /jira/secure/useravatar?size=xsmall&amp;ownerId=rs&amp;avatarId=10500 HTTP/1.1" 404 2293 0 "-" "mattermost-5.10.0" "-"
192.168.2.88 707x425015x1 - [02/May/2019:11:47:53 +0200] "GET /jira/secure/viewavatar?size=xsmall&amp;avatarId=10800&amp;avatarType=issuetype HTTP/1.1" 404 2293 15 "-" "mattermost-5.10.0" "-"
192.168.2.88 707x425009x1 - [02/May/2019:11:47:53 +0200] "GET /jira/secure/projectavatar?size=medium&pid=11000&avatarId=11003 HTTP/1.1" 403 19653 31 "-" "mattermost-5.10.0" "-"

So it's either a 404, possible due to a malformed URL (&amp; should be &) or a 403 (Last log entry).
I checked that the nginx configuration: It's configured as described in the Mattermost manual.

As the URLs work when called on the Mattermost server via curl:
`curl https://connect.schaeferbarthold.de/jira/secure/projectavatar?size=medium&pid=11000&avatarId=11003`

Result:
`<svg id="Default" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 128 128"><style>.st1{fill:#4a2593}.st3{opacity:.15;fill:#7f4cbf;enable-background:new}.st4{fill:#fff}</style><path d="M116 128H12c-6.6 0-12-5.4-12-12V12C0 5.4 5.4 0 12 0h104c6.6 0 12 5.4 12 12v104c0 6.6-5.4 12-12 12z" fill="#00a3bf"/><path class="st1" d="M40.3 109.7h-2.1c-1.8 0-3.2-1.4-3.2-3.2V93.9c0-1.8 1.4-3.2 3.2-3.2h2.1c1.8 0 3.2 1.4 3.2 3.2v12.5c0 1.8-1.5 3.3-3.2 3.3zm48.7 0h-2.1c-1.8 0-3.2-1.4-3.2-3.2V93.9c0-1.8 1.4-3.2 3.2-3.2H89c1.8 0 3.2 1.4 3.2 3.2v12.5c0 1.8-1.4 3.3-3.2 3.3z"/><ellipse class="st1" cx="63.8" cy="74.7" rx="43.6" ry="30"/><ellipse cx="64" cy="67.7" rx="50.3" ry="30" fill="#7f4cbf"/><path class="st3" d="M32.5-57.2c0 .5-.1.9-.1 1.4v8.3C32.9-30.8 46.6-23 63.6-23c17.2 0 31.2-8.1 31.2-25.4 0-.6 0-1.3-.1-1.9v-2.6c0-.6.1-1.3.1-1.9 0-.9 0-1.9-.1-2.8-8.5-3.9-19.2-6.3-30.7-6.3-12 0-22.9 2.5-31.5 6.7z"/><path class="st4" d="M63.8 15.5c-.1 0-.1 0 0 0-17 0-30.8 13.4-31.3 30.2V54c.3 9.7 5 16.3 12.3 20.3 5.2 2.8 11.8 4.2 18.8 4.2s13.6-1.4 18.8-4.2c7.3-4 12.3-10.6 12.6-20.3v-8.3c-.5-16.8-14.3-30.2-31.2-30.2z"/><path d="M63.6 28.3c-10.4 0-18.8 8.5-18.8 18.8v27.2c5.2 2.8 11.8 4.2 18.8 4.2s13.6-1.4 18.8-4.2V47.1c0-10.4-8.4-18.8-18.8-18.8z" fill="#59afe1"/><circle class="st4" cx="63.9" cy="50.7" r="11.1"/><circle cx="63.5" cy="50.6" r="4.4" fill="#2a5083"/><path class="st3" d="M94.9 44c-8.5-4-19.2-6.3-30.9-6.3-11.9 0-22.8 2.5-31.4 6.6 0 .5-.1.9-.1 1.4V54c.3 9.7 5 16.3 12.3 20.3 5.2 2.8 11.8 4.2 18.8 4.2s13.6-1.4 18.8-4.2c7.3-4 12.3-10.6 12.6-20.3v-8.3c0-.6-.1-1.1-.1-1.7z"/></svg>`

I assume that maybe the accept header property is not set to accept a svg file.

**Comment by jprusch** [comment]
Another update:
The issue appears only, if the local image proxy is enabled. Switching off the image proxy, displays the images & avatars when using the browser client. This is the expected behavior. Anyway, the image proxy is needed, when using the desktop or mobile apps.

**Comment by towolf** [comment]
We have the same issue. I change the settings to `0.0.0.0/0`, but that is not a good solution I think.

**Comment by jwilander** [comment]
I spent some time trying to reproduce this on latest master but haven't been able to. It seems to work fine for me when setting AllowUntrustedInternalConnections to `madeuphost` which in my hosts file I've pointed to 127.0.0.1. Serving an image on my localhost, with the proxy enabled `http://madeuphost/img/img.png` fails to resolve with the correct error if I don't have AllowUntrustedInternalConnections set. When I do set AllowUntrustedInternalConnections, I am able to resolve the image and have it displayed by Markdown in channel.

Is anyone still running into this on 5.18?
